src/backend/distributed/transaction/backend_data.c

```diff
@@ -566,3 +573,70 @@ GetBackendDataForProc(PGPROC *proc, BackendData *result)
 
 	SpinLockRelease(&backendData->mutex);
 }
+
+
+/*
+ * CancelTransactionDueToDeadlock cancels the input proc and also marks the backend
+ * data with this information.
+ */
+void
+CancelTransactionDueToDeadlock(PGPROC *proc)
+{
+	BackendData *backendData = &backendManagementShmemData->backends[proc->pgprocno];
+
+	/* backend might not have started */
```

Comment:
'not started' doesn't seem right. Maybe 'might not have used citus yet and thus not initialized backend data'?